src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaFallBackProducer.java

```diff
@@ -0,0 +1,53 @@
+package com.zendesk.maxwell.producer;
+
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Properties;
+
+public class MaxwellKafkaFallBackProducer {
+	private static final Logger LOGGER = LoggerFactory.getLogger(MaxwellKafkaFallBackProducer.class);
+
+	private static MaxwellKafkaFallBackProducer fallBackProducer;
+
+	public static MaxwellKafkaFallBackProducer getInstance(Properties kafkaProperties) {
+		if (fallBackProducer == null) {
+			synchronized (MaxwellKafkaFallBackProducer.class) {
+				if (fallBackProducer == null) {
+					fallBackProducer = new MaxwellKafkaFallBackProducer(kafkaProperties);
+				}
+			}
+		}
+
+		return fallBackProducer;
+	}
```

Comment:
Singleton pattern is used to make sure we only need one producer for all the oversized messages. Otherwise, creating a producer for each message seems costly.

Reply:
We only have one instance of `MaxwellKafkaProducer` per process, we could instantiate the fallback producer at the same time (either when we build the regular producer, or in the producer's constructor)

Reply:
That'll still require to pass the fallback producer around to every callback? Which is not nice.

Reply:
It's not beautiful, but it's simple 🤷‍♂️ - we already pass around a bunch of other things, adding one more argument is less complexity than this code.